grovedb/src/replication.rs

```diff
@@ -402,4 +462,124 @@ pub(crate) mod utils {
         }
         Ok(res)
     }
+
+    /// Packs a vector of byte vectors (`Vec<Vec<u8>>`) into a single byte
+    /// vector.
+    ///
+    /// The encoding format is as follows:
+    /// 1. The first byte stores the number of elements.
+    /// 2. Each element is prefixed with its length as a 2-byte (`u16`) value in
+    ///    big-endian format.
+    /// 3. The actual byte sequence of the element is then appended.
+    ///
+    /// # Arguments
+    ///
+    /// * `nested_bytes` - A vector of byte vectors (`Vec<Vec<u8>>`) to be
+    ///   packed.
+    ///
+    /// # Returns
+    ///
+    /// A `Vec<u8>` containing the encoded representation of the nested byte
+    /// vectors.
+    pub fn pack_nested_bytes(nested_bytes: Vec<Vec<u8>>) -> Vec<u8> {
```

Comment:
vector of vectors is a pretty strict requirement, however, what actually used here is a sequence of slices and could be lifted to something like `impl IntoIter<Item = &[u8]>`

this isn't a dealbreaker for me of course, but if somewhere in the code it requires `.to_vec()` conversions to match this requirement then consider this change